### PR TITLE
Added docstring as test case text

### PR DESCRIPTION
**Pull request by jcm-mx**
Making it so the text in the docstrings is what gets reported as text in kiwi tcms. This should allow for more descriptive test cases in kiwi

### CODE REVIEW

File: tests/test_docstring_reporting.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024
```
Comment (atodorov) [inline tests/test_docstring_reporting.py]: Please use 2026 as the year and also add your name/email as the author. Feel free to add yourself to `__init__.py` as well.

File: tests/test_docstring_reporting.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    plugin.pytest_collection_finish(session)  # must not raise
+
+    plugin.backend.update_test_case_text.assert_not_called()
+
+
+def test_docstring_not_sent_for_different_nodeid(plugin):
+    """Docstring harvested for test A must not be sent when test B runs."""
+    item_a = _make_item("tests/test_foo.py::test_a", "Intention A")
+    item_b = _make_item("tests/test_foo.py::test_b", None)
+    item_b.function.__doc__ = None
```
Comment (atodorov) [inline tests/test_docstring_reporting.py]: Q: If `_make_item()` takes `None` as an argument, why do you need to set the docstring explicitly on line 114 ?

File: tests/test_docstring_reporting.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    _simulate_collection(plugin, [item_a, item_b])
+
+    _simulate_logstart(plugin, "tests/test_foo.py::test_b")
+
+    plugin.backend.update_test_case_text.assert_not_called()
```
Comment (atodorov) [inline tests/test_docstring_reporting.py:129]: I think a better assertion here would be that we've updated the text for test_a but not for test_b. Otherwise we may be missing a corner case.

File: tcms_pytest_plugin/__init__.py [hunk 4]
```diff
@@ -58,6 +73,10 @@ def pytest_runtest_logstart(self, nodeid, location):
             test_case["id"], self.backend.run_id
         )
 
+        docstring = self._docstrings.get(nodeid)
+        if docstring:
+            self.backend.update_test_case_text(test_case["id"], docstring)
```
Comment (atodorov) [inline tcms_pytest_plugin/__init__.py]: Please move this immediately after line 70 - it makes more logical sense there.

File: tcms_pytest_plugin/__init__.py [hunk 5]
```diff
@@ -58,6 +73,10 @@ def pytest_runtest_logstart(self, nodeid, location):
             test_case["id"], self.backend.run_id
         )
 
+        docstring = self._docstrings.get(nodeid)
+        if docstring:
```
Comment (atodorov) [inline tcms_pytest_plugin/__init__.py]: Update only if text is different:

```suggestion
        if docstring != test_case['text']:
```

IOW source code has changed or (most likely) the DB record is newly created!

**Review by atodorov — CHANGES_REQUESTED** [review]
Overall looks good, needs a few small changes.

File: tests/test_docstring_reporting.py (hunk 1 above)
Comment (atodorov) [reply tests/test_docstring_reporting.py]: @jcm-mx , @cm-dynamic the copyright notice needs a proper authorship attribution, which looking at the git log is both of you. See `__init__.py` for example.

File: tests/test_docstring_reporting.py (hunk 3 above)
Comment (atodorov) [reply tests/test_docstring_reporting.py:129]: @jcm-mx , @cm-dynamic any news on this one? Test could be improved with stronger assertions.

File: tests/test_docstring_reporting.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2026
+# Licensed under the GPLv3: https://www.gnu.org/licenses/gpl.html
+#
+# pylint: disable=redefined-outer-name
+
+from unittest.mock import MagicMock, patch
```
Comment (atodorov) [inline tests/test_docstring_reporting.py:7]: See the bottom of `.github/workflows/testing.yml` and update the grep call with the expected number of test cases/statuses: 

```
1 failed, 8 passed, 1 skipped, 1 error
```

**Review by atodorov — CHANGES_REQUESTED** [review]
LGTM, small nitpicks remaining.

File: tests/test_docstring_reporting.py (hunk 1 above)
Comment (jcm-mx) [reply tests/test_docstring_reporting.py]: @atodorov  Fixed, it's the same person for both handlers

File: tests/test_docstring_reporting.py (hunk 3 above)
Comment (jcm-mx) [reply tests/test_docstring_reporting.py:129]: I think the assertions are better now

File: tests/test_docstring_reporting.py (hunk 6 above)
Comment (jcm-mx) [reply tests/test_docstring_reporting.py:7]: Done

**Review by atodorov — APPROVED** [review]
